Topics/System_Design/Security/Secure_Boot_RSA.md

```diff
@@ -0,0 +1,101 @@
+
+For a more in depth explanation of how the entire secure boot process works, feel free to consult external resources such as the provided link 
+[here](https://www.particle.io/iot-guides-and-resources/iot-secure-boot/#:~:text=In%20the%20context%20of%20Internet,at%20the%20time%20of%20manufacture)
+
+
+## What is RSA?
+RSA (Rivest–Shamir–Adleman) is an asymmetric encryption algorithm, meaning it uses a pair of keys where one key is used for encryption and one key is used for decryption. The security of RSA relies on the difficulty of factoring the product of two large prime numbers, making it computationally impossible to derive the private key from the public key.
+
+![Diagram explaining Asymmetric Encryption](../assets/key.png)
+
+### Asymmetric Encryption versus Symmetric Encryption
+
+
+In symmetric encryption, a single key is used for both encryption and decryption, rather than a public private key pair. This means that the same key is used by both the sender and the recipient to secure and access the information.
+
+It is a lot faster than asymmetric encryption and computationally less intensive.
+The challenge in symmetric encryption lies in securely distributing and managing the shared secret key. If a third party gains access to the key, they can decrypt all the communication.
+
+An example of an asymmetric encryption algorithm is the [Advanced Encryption Standard (AES)](https://www.techtarget.com/searchsecurity/definition/Advanced-Encryption-Standard)
+
+![Diagram explaining Symmetric Encryption](../assets/symmetric.png)
+
+
+### The Math behind RSA
+
+#### Key Generation
+1. Choose two distinct prime numbers, p = 11 and q = 17.
+
+2. Calculate n = p x q, in this case 11 x 17 = 187. This term is called the modulus
+
+3. Calculate the Eulers Toient Function ϕ(n) = (p-1) x (q-1) = 10 x 16 = 160
+
+4. Choose a public exponent e such that 1<e<ϕ(n) and e is coprime with ϕ(n). Let's choose e=7.
+
+5. Calculate the private exponent d such that d≡e^(mod ϕ(n)). In this case, d≡23.
+
+6. The public key is (e, n) and the private key is (d, n). Where e = 7, n = 187, d = 23
+
+#### Encryption
+
+1. Allen wants to send a message like M=88 to Kai. Kais public key is (e,n) = (7, 187)
+
+2. Compute the ciphertext C using C≡M^(e)(modn). So, C≡88^ (7) (mod187) which is 11.
+3. The calculated ciphertext is sent to Kai using his public key
+
+#### Decryption
+
+1. The calculated Ciphertext C = 11 is sent to Kai
+2. Kai will use his private key (d,n) = (23, 187) to decrypt M ≡ C^(d)(mod n). So now we have that M ≡ 11^(23) (mod 187)
+3. The result is M = 88 which is the original message we sent!
+
+
+##### This is just a simple explanation of the math behind RSA, and in reality much larger numbers like 2048, and 4096-bit numbers are often used to guarantee that the private key cannot be derived from the public key
+
+
+## RSA Usage
```

Comment:
I think it would be nice if you could give a brief summary at the end of each ## heading section